addons/account/models/account_move.py

```diff
@@ -3086,6 +3086,10 @@ def _autopost_draft_entries(self):
             ('state', '=', 'draft'),
             ('date', '<=', fields.Date.context_today(self)),
             ('auto_post', '=', True),
+            ('invoice_line_ids', '!=', False),
+            '|',
+            ('move_type', '=', 'entry'),
+            ('partner_id', '!=', False),
```

Comment:
to me, you need to catch an `UserError` and try to exclude the problematic moves instead of trying to handle everything in advance with a domain